rio_tiler/utils.py

```diff
@@ -121,21 +122,36 @@ def get_array_statistics(
         else:
             percentiles_values = (numpy.nan,) * len(percentiles_names)
 
+        if coverage is not None:
+            assert coverage.shape == (
+                data.shape[1],
+                data.shape[2],
+            ), f"Invalid shape ({coverage.shape}) for Coverage, expected {(data.shape[1], data.shape[2])}"
+
+            array = data[b] * coverage
+
+        else:
+            array = data[b]
+
+        count = array.count() if coverage is None else coverage.sum()
+        if valid_pixels:
+            majority = float(keys[counts.tolist().index(counts.max())].tolist())
+            minority = float(keys[counts.tolist().index(counts.min())].tolist())
+        else:
+            majority = numpy.nan
+            minority = numpy.nan
+
         output.append(
             {
                 "min": float(data[b].min()),
                 "max": float(data[b].max()),
-                "mean": float(data[b].mean()),
-                "count": float(data[b].count()),
-                "sum": float(data[b].sum()),
-                "std": float(data[b].std()),
-                "median": float(numpy.ma.median(data[b])),
-                "majority": float(keys[counts.tolist().index(counts.max())].tolist())
-                if valid_pixels
-                else numpy.nan,
-                "minority": float(keys[counts.tolist().index(counts.min())].tolist())
-                if valid_pixels
-                else numpy.nan,
+                "mean": float(array.mean()),
```

Comment:
we don't use the weighted array for the `min` and `max` (like https://github.com/isciences/exactextract#supported-statistics)